### ISSUE TITLE
BUG: Missing sigmoid in GroundingSAM2 postprocess & proposed solution

### ISSUE THREAD

**Issue by tiucamaria** [CVHub520/X-AnyLabeling#1179]
### Search before asking

- [x] I have searched the X-AnyLabeling [issues](https://github.com/CVHub520/X-AnyLabeling/issues) and found no similar bug report.

### Bug

Hello!
I was using your project to detect and segment cars in the UAVID dataset using GroundingSAM2, but I wasn't satisfied with the result, so I decided to investigate further. I tried one of your old examples in the README file so I could compare the outputs, but they weren't the same.
Right now, if you run the GroundingSAM2 model on this image, the output will be:

<img width="1920" height="992" alt="Image" src="https://github.com/user-attachments/assets/4df3c382-bcab-4241-b907-49a73afe9682" />
<img width="1920" height="995" alt="Image" src="https://github.com/user-attachments/assets/3dfcc99c-cfae-47c7-a779-475385473911" />

Next, I checked which objects were being detected, and here I found the problem: the exact DINO model used in GorundingSAM2 is the same as the one that produces these results:

<img width="1920" height="1027" alt="Image" src="https://github.com/user-attachments/assets/f4244829-b28c-4bdd-b28a-2bf60ab93eb9" />
<img width="1920" height="1030" alt="Image" src="https://github.com/user-attachments/assets/08f93762-23fd-481c-a111-48853cc8f633" />

Then, I tested it again on one of my UAVID images, where I knew some objects should be detected. I printed the scores and noticed that the predictions returned values of 1.x instead of being in the range (0, 1). In the GroundingSAM2 file is omitted to apply the sigmoid (sig function). Here is how I modified the code:

def postprocess(
self, outputs, caption, with_logits=True, token_spans=None
):
logits, boxes = outputs
# logits_filt = np.squeeze(
# logits, 0
# ) # [0] # prediction_logits.shape = (nq, 256)
logits_filt = self.sig(np.squeeze(logits, 0)) # BUG SOLVED
......

See the results after adding the call of the sig function:

<img width="1920" height="1030" alt="Image" src="https://github.com/user-attachments/assets/4adada63-75d4-461c-ac49-4eb340cb9cbe" />
<img width="1917" height="1030" alt="Image" src="https://github.com/user-attachments/assets/631f1503-121e-4f98-8ad1-0cb76cd07bd1" />

### Information

Application Information:
    {'App name': 'X-AnyLabeling', 'App version': '3.0.3', 'Device': 'GPU'}

System Information:
    {'CPU': 'Intel64 Family 6 Model 186 Stepping 2, GenuineIntel',
     'CUDA': None,
     'GPU': '0, NVIDIA GeForce RTX 4060 Laptop GPU, 8188',
     'Operating System': 'Windows-10-10.0.19045-SP0',
     'Python Version': '3.10.18'}

Package Information:
    {'ONNX Runtime GPU Version': '1.22.0',
     'ONNX Runtime Version': None,
     'ONNX Version': '1.16.1',
     'OpenCV Contrib Python Headless Version': '4.11.0.86',
     'PyQt5 Version': '5.15.7'}

### Link to a Reproducible Demonstration Video

https://drive.google.com/file/d/1ibhkNuyp0-mLEEtvInq59cDcQl4HJtP-/view?usp=sharing

### Execution Mode

Source Code

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by CVHub520** [comment]
Hey there! @tiucamaria, thanks for reporting this issue! We appreciate you bringing it to our attention.

Good news! This has been addressed and fixed in the latest version of X-AnyLabeling. We strongly recommend updating to the newest branch, as it includes many improved features and a more robust operational mechanism, which we believe will greatly enhance your experience.

Happy labeling! 🚀
